packages/kbn-es-archiver/src/lib/docs/generate_doc_records_stream.test.ts

```diff
@@ -20,48 +20,24 @@ import { createStats } from '../stats';
 
 const log = new ToolingLog();
 
-it('transforms each input index to a stream of docs using scrollSearch helper', async () => {
-  const responses: any = {
-    foo: [
-      {
-        body: {
-          hits: {
-            total: 5,
-            hits: [
-              { _index: 'foo', _type: '_doc', _id: '0', _source: {} },
-              { _index: 'foo', _type: '_doc', _id: '1', _source: {} },
-              { _index: 'foo', _type: '_doc', _id: '2', _source: {} },
-            ],
-          },
-        },
-      },
-      {
-        body: {
-          hits: {
-            total: 5,
-            hits: [
-              { _index: 'foo', _type: '_doc', _id: '3', _source: {} },
-              { _index: 'foo', _type: '_doc', _id: '4', _source: {} },
-            ],
-          },
-        },
-      },
-    ],
-    bar: [
-      {
-        body: {
-          hits: {
-            total: 2,
-            hits: [
-              { _index: 'bar', _type: '_doc', _id: '0', _source: {} },
-              { _index: 'bar', _type: '_doc', _id: '1', _source: {} },
-            ],
-          },
-        },
-      },
-    ],
-  };
+interface SearchResponses {
+  [key: string]: Array<{
+    body: {
+      hits: {
+        total: number;
+        hits: Array<{
+          _index: string;
+          _type: string;
+          _id: string;
+          _source: Record<string, unknown>;
+        }>;
+      };
+    };
+  }>;
+}
 
+function createMockClient(responses: SearchResponses) {
+  // TODO: replace with proper mocked client
   const client: any = {
     helpers: {
       scrollSearch: jest.fn(function* ({ index }) {
```

Comment:
I abstracted this out into a reusable function, but it could/should be improved so I left a TODO here. I wanted to get this PR done with minimal changes so I opted not to change this existing mock logic.